src/messages/mod.rs

```diff
@@ -170,15 +170,16 @@ impl Message {
     }
 
     /// Verify this message is properly signed and trusted.
-    pub(crate) fn verify<'a, I>(&'a self, trusted_keys: I) -> Result<VerifyStatus>
+    pub(crate) fn verify<'a, I>(&'a self, trusted_keys: I) -> Result<(), IntegrityError>
     where
         I: IntoIterator<Item = (&'a Prefix, &'a bls::PublicKey)>,
     {
         let bytes = bincode::serialize(&SignableView {
             dst: &self.dst,
             dst_key: self.dst_key.as_ref(),
             variant: &self.variant,
-        })?;
+        })
+        .map_err(|_| IntegrityError::FailedSignature)?;
```

Comment:
Yes but we serialize it only so we can calculate the signature from it. So failure to serialize here results in failure to verify. We already do something like this in other places, for example `Proven::self_verify` returns `false` even on serialize failure.

That said, I don't have a strong opinion on this. So if you think we should rather return a serialization error then I don't mind that much changing it.

Reply:
Well, as it's aimed to make the returned error more accurate, I think it will help if return with a serialisation error.
Which may help the caller to carry out restore work ?